clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -473,7 +473,7 @@ private void transitionTo(State target) {
 
     private synchronized void transitionTo(State target, RuntimeException error) {
         if (!currentState.isTransitionValid(currentState, target))
-            throw new KafkaException("Invalid transition attempted from state " + currentState.name() +
+            throw new KafkaException(logPrefix + "Invalid transition attempted from state " + currentState.name() +
```

Comment:
A little odd to include the log prefix in an exception message. Perhaps we could add the transactionalId to the exception message directly?